webui/react/src/pages/F_ExpList/glide-table/ColumnPickerMenu.tsx

```diff
@@ -0,0 +1,197 @@
+      (acc, col) => ({ ...acc, [col.column]: !allFilteredColumnsChecked }),
+      {},
+    );
+
+    setColumnState((prevCols) => ({ ...prevCols, ...filteredColumnMap }));
+  }, [allFilteredColumnsChecked, filteredColumns, setColumnState]);
+
+  const handleColumnChange = useCallback(
+    (event: CheckboxChangeEvent) => {
+      const [id, checked] = [event.target.id, event.target.checked];
+      if (id === undefined) return;
+      setColumnState((prevState) => ({ ...prevState, [id]: checked }));
+    },
+    [setColumnState],
+  );
+
+  const handleSearch = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setSearch(e.target.value);
+    },
+    [setSearch],
+  );
+
+  return (
+    <div>
+      <Input allowClear autoFocus placeholder="Search" value={search} onChange={handleSearch} />
+      {totalColumns.length !== 0 ? (
+        <Space className={css.columnList} direction="vertical">
+          {filteredColumns.length > 0 ? (
+            filteredColumns.map((col) => (
+              <Checkbox
+                checked={columnState[col.column] ?? false}
+                id={col.column}
+                key={col.column}
+                onChange={handleColumnChange}>
+                {col.displayName || col.column}
+              </Checkbox>
+            ))
+          ) : (
+            <Empty description="No results" />
+          )}
+        </Space>
+      ) : (
+        <Spinner />
+      )}
+      <div className={css.actionRow}>
+        <Button type="text" onClick={handleShowHideAll}>
+          {allFilteredColumnsChecked ? 'Hide' : 'Show'} all
+        </Button>
+        <Button type="text" onClick={handleShowSuggested}>
+          Reset
+        </Button>
+      </div>
+    </div>
+  );
+};
+
+const ColumnPickerMenu: React.FC<ColumnMenuProps> = ({
+  projectColumns,
+  setVisibleColumns,
+  initialVisibleColumns,
+}) => {
+  const [search, setSearch] = useState('');
+
+  const totalColumns = useMemo(
+    () => removeBannedColumns(Loadable.getOrElse([], projectColumns)),
+    [projectColumns],
+  );
+
+  const [columnState, setColumnState] = useState<Record<string, boolean>>(() =>
+    totalColumns.reduce(
+      (acc, col) => ({ ...acc, [col.column]: initialVisibleColumns.includes(col.column) }),
+      {},
+    ),
+  );
+
+  useEffect(() => {
+    if (Object.keys(columnState).length === 0) return;
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    setVisibleColumns(
+      Object.entries(columnState)
+        .filter(([_, checked]) => checked)
+        .map(([column, _]) => column),
+    );
+    /* eslint-enable @typescript-eslint/no-unused-vars */
+  }, [columnState, setVisibleColumns]);
+
+  const handleShowSuggested = useCallback(() => {
+    const defaultCols: Set<string> = new Set(defaultExperimentColumns);
+
+    setColumnState((prevColumns) =>
+      Object.fromEntries(Object.keys(prevColumns).map((col) => [col, defaultCols.has(col)])),
+    );
+  }, []);
```

Comment:
Alright, I made that change. Once small... regression? maybe? is that now if you remove and re-add a column it will show up at the end of the table instead of reappearing where it was.